src/storage/heap_file.c

```diff
@@ -18935,6 +18894,7 @@ heap_set_mvcc_rec_header_on_overflow (PAGE_PTR ovf_page, MVCC_REC_HEADER * mvcc_
 
   assert (ovf_page != NULL);
   assert (mvcc_header != NULL);
+  assert (MVCC_IS_FLAG_SET (mvcc_header, OR_MVCC_FLAG_VALID_DELID));
```

Comment:
See the comment below: `/* Make sure the header has maximum size for overflow records */`.
This means it is expected to receive a header without the flag set and you have to handle the case like it's done for ins_id.